addons/calendar/controllers/main.py

```diff
@@ -91,3 +91,14 @@ def notify(self):
     @http.route('/calendar/notify_ack', type='json', auth="user")
     def notify_ack(self):
         return request.env['res.partner'].sudo()._set_calendar_last_notif_ack()
+
+    @http.route('/calendar/meeting/join', type='http', auth="user", website=True)
+    def calendar_join_meeting(self, token, **kwargs):
```

Comment:
I would move it below view_meeting, so that we let JSON routes at the end of the file and keep http / json separated, like well ranged chaussettes.